distribution/lambda/Makefile

```diff
@@ -0,0 +1,127 @@
+.SILENT:
+.ONESHELL:
+SHELL := bash
+.SHELLFLAGS := -eu -o pipefail -c
+
+QW_LAMBDA_VERSION?=beta
+PACKAGE_BASE_URL=https://github.com/quickwit-oss/quickwit/releases/download/aws-lambda-$(QW_LAMBDA_VERSION)/
+SEARCHER_PACKAGE_FILE=quickwit-lambda-searcher-$(QW_LAMBDA_VERSION)-x86_64.zip
+INDEXER_PACKAGE_FILE=quickwit-lambda-indexer-$(QW_LAMBDA_VERSION)-x86_64.zip
+export SEARCHER_PACKAGE_PATH=cdk.out/$(SEARCHER_PACKAGE_FILE)
+export INDEXER_PACKAGE_PATH=cdk.out/$(INDEXER_PACKAGE_FILE)
+
+check-env:
+ifndef CDK_ACCOUNT
+	$(error CDK_ACCOUNT is undefined)
+endif
+ifndef CDK_REGION
+	$(error CDK_REGION is undefined)
+endif
+
+# Build or download the packages from the release page
+# - Download by default, the version can be set with QW_LAMBDA_VERSION
+# - To build locally, set QW_LAMBDA_BUILD=1
+package:
+	mkdir -p cdk.out
+	if [ "$${QW_LAMBDA_BUILD:-0}" = "1" ]
+	then
+	  pushd ../../quickwit/
+	  cargo lambda build \
+	    -p quickwit-lambda \
+	    --release \
+	    --output-format zip \
+	    --target x86_64-unknown-linux-gnu
+	  popd
+	  cp -u ../../quickwit/target/lambda/searcher/bootstrap.zip $(SEARCHER_PACKAGE_PATH)
+	  cp -u ../../quickwit/target/lambda/indexer/bootstrap.zip $(INDEXER_PACKAGE_PATH)
+	else
+	  if ! [ -f $(SEARCHER_PACKAGE_PATH) ]; then
+	    curl -C - -Ls -o $(SEARCHER_PACKAGE_PATH) $(PACKAGE_BASE_URL)$(SEARCHER_PACKAGE_FILE)
+	  fi
+	  if ! [ -f $(INDEXER_PACKAGE_PATH) ]; then
+	  	curl -C - -Ls -o $(INDEXER_PACKAGE_PATH) $(PACKAGE_BASE_URL)$(INDEXER_PACKAGE_FILE)
+	  fi
+	fi
+
+indexer-package-path:
+	echo -n $(INDEXER_PACKAGE_PATH)
+
+searcher-package-path:
+	echo -n $(SEARCHER_PACKAGE_PATH)
+
+bootstrap: package check-env
+	cdk bootstrap aws://$$CDK_ACCOUNT/$$CDK_REGION
+
+deploy-hdfs: package check-env
+	cdk deploy -a cdk/app.py HdfsStack
+
+deploy-mock-data: package check-env
+	cdk deploy -a cdk/app.py MockDataStack
+
+destroy-hdfs:
+	cdk destroy -a cdk/app.py HdfsStack
+
+destroy-mock-data:
+	cdk destroy -a cdk/app.py MockDataStack
+
+clean:
+	rm -rf cdk.out
+
+## Invocation examples
+
+invoke-mock-data-searcher: check-env
+	python -c 'from cdk import cli; cli.invoke_mock_data_searcher()'
+
+invoke-hdfs-indexer: check-env
+	python -c 'from cdk import cli; cli.upload_hdfs_src_file()'
+	python -c 'from cdk import cli; cli.invoke_hdfs_indexer()'
+
+invoke-hdfs-searcher-term: check-env
+	python -c 'from cdk import cli; cli.invoke_hdfs_searcher("""{"query": "severity_text:ERROR", "max_hits": 10}""")'
+
+invoke-hdfs-searcher-histogram: check-env
+	python -c 'from cdk import cli; cli.invoke_hdfs_searcher("""{ "query": "*", "max_hits": 0, "aggs": { "events": { "date_histogram": { "field": "timestamp", "fixed_interval": "1d" }, "aggs": { "log_level": { "terms": { "size": 10, "field": "severity_text", "order": { "_count": "desc" } } } } } } }""")'
+
+bench-index:
```

Comment:
Leaving the benchmark scripts as part of the code base. We might want to remove them to avoid clutter and improve readability.

Reply:
Yes, after playing a bit with cdk, I think the best is to keep in the repository only the stuff related to the build of the binaries and move everything else in a dedicated repository.
We can do this cleanup just after the release.